server/testutil/testenv/testenv.go

```diff
@@ -135,26 +139,32 @@ func GetTestEnv(t testing.TB) *TestEnv {
 	if err != nil {
 		t.Fatal(err)
 	}
-	configurator, err := config.NewConfigurator(tmpConfigFile)
-	if err != nil {
-		t.Fatal(err)
+	if currentConfigurator == nil {
+		configurator, err := config.NewConfigurator(tmpConfigFile)
+		if err != nil {
+			t.Fatal(err)
+		}
+		currentConfigurator = configurator
+		t.Cleanup(func() {
+			currentConfigurator = nil
+		})
```

Comment:
I don't know if you need to worry about race conditions here, but a once.Do() (from sync) might be a more idiomatic way to initialize this?

Reply:
I don't think we need to worry about that. We only perform component setup in the main goroutine.

It does mean `t.Parallel()` won't work properly but it already doesn't work because of the shared config struct.
Getting `t.Parallel()` to work would require larger changes.